datadog/provider.go

```diff
@@ -100,6 +103,12 @@ type ProviderConfiguration struct {
 	DatadogClientV2 *datadogV2.APIClient
 	AuthV1          context.Context
 	AuthV2          context.Context
+
+	now func() time.Time
```

Comment:
Why not directly?
```suggestion
	Now func() time.Time
```

Reply:
I didn't even think of that haha. Guess it felt more "go" the other way, but yeah, I guess this works too